TypeCobol/Compiler/Nodes/Data.cs

```diff
@@ -957,6 +957,17 @@ public override DataType DataType
             get { return DataType.Numeric; }
         }
 
+        public override DataUsage? Usage
+        {
+	        get { return DataUsage.Index; }
+        }
+
+        public override long PhysicalLength
+        {
+            //Table indexes are not stored along their declaring parent, so they do not contribute to data item length.
+            get { return 0; }
```

Comment:
I am doubtful on this length of 0.
It seems strange to have an actual variable (as you can use SET on it) with a size of 0 in memory.
Maybe you can remove it from the computation of the length of the defining array and give it a real size?

Reply:
This is indeed a bit misleading. After discussing it with @sarterbe, he confirms that a zero `PhysicalLength` for a table index is more correct than a value (4 in our case) because table indexes are privately managed by the compiler. Their size and location in memory are implementation details that we have no control over and that we should never consider.

As a consequence, the properties `PhysicalLength`, `StartPosition`, `PhysicalPosition` and `SlackBytes` should throw an error when called on an `IndexDefinition` instance (or return `null` if we change their return type to a nullable). As this is a bigger change in the code and that we do not really require the functionality, we'll keep this 'trick' for now.

Reply:
Yes, agreed with that approach.